src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -0,0 +1,84 @@
+/**
+Copyright 2018 BarD Software s.r.o
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.bardsoftware.papeeria.backend.cosmas
+
+import com.google.cloud.storage.*
+import io.grpc.stub.StreamObserver
+import com.google.cloud.storage.Acl.User
+import com.google.cloud.storage.contrib.nio.testing.LocalStorageHelper
+import com.google.protobuf.ByteString
+import io.grpc.Status
+import io.grpc.StatusException
+import java.util.Arrays
+import java.util.ArrayList
+
+/**
+ * Special class that can work with requests from CosmasClient
+ * This realization stores files in Google Cloud Storage.
+ *
+ * @author Aleksandr Fedotov (iisuslik43)
+ */
+class CosmasGoogleCloudService(private val bucketName: String) : CosmasGrpc.CosmasImplBase() {
+
+    private var storage: Storage = StorageOptions.getDefaultInstance().service
```

Comment:
This can be `val`, I believe.

Reply:
No, it works only with `var` because I change this field in the second constructor

Reply:
That's another reason to move it into primary constructor args.